File: leetcode/easy/26_remove_duplicates_from_sorted_array/mac.js
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/
// Runtime: 80 ms, faster than 55.95% of JavaScript online submissions for Remove Duplicates from Sorted Array.
// Memory Usage: 37.6 MB, less than 25.00% of JavaScript online submissions for Remove Duplicates from Sorted Array.

/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    const main = {
        init : function(e) {
            for (let i = 1; i < e.length; i++) {
                if (e[i-1] === e[i]) {
                    e.splice(i, 1);
                    i--;
                }
            }
            
            return e;
        }
    }
    
    main.init(nums);
    
};

File: leetcode/easy/26_remove_duplicates_from_sorted_array/mac.swift
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/
// Runtime: 76 ms, faster than 77.30% of Swift online submissions for Remove Duplicates from Sorted Array.
// Memory Usage: 20.9 MB, less than 25.00% of Swift online submissions for Remove Duplicates from Sorted Array.

class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        if nums.count > 1 { 
            var count = 1

            for i in 1..<nums.count {
                if nums[i] != nums[i-1] {
                    nums[count] = nums[i]
                    count += 1
                }
            }

            return count
        } else {
            return nums.count
        }
        
    }
}